Core/Source/MVVM/FilteredModelCollection.swift

```diff
@@ -124,69 +111,32 @@ public final class FilteredModelCollection: ModelCollection, ProxyingCollectionE
         filterCookie += 1
     }
 
-    /// Cache of the results of the last filter pass for each item
-    private var filterResults: [[Bool]]?
-
     private func runFilter() {
         assert(Thread.isMainThread)
 
         let originalSections = sourceCollection.sections
         let originalFilter = filter
         let originalCookie = filterCookie
-        let originalIncluded = filterResults
         let originalLimit = limit
 
         state = .loading(state.sections)
 
-        var newFilterResults: [[Bool]] = []
         var newFilteredSections: [[Model]] = []
-        var insertedIndexPaths: [ModelPath] = []
-        var removedIndexPaths: [ModelPath] = []
 
         let filterCollection = {
             var itemCount = 0
-            // Core filtering work.
-            for (sectionIndex, sectionModels) in originalSections.enumerated() {
-                // To avoid making multiple passes through the collection, we compare filter results to a saved copy of
-                // the results from the previous filter (when that exists). As we increment through the models we keep a
-                // pointer to the current index to insert the next row if its included in the new filter, and not the
-                // old one. As well as a pointer that increments for every row included previously so that we can delete
-                // the row for the current item if its not included by the new filter but was by the old one.
-                var insertionIndex = 0
-                var deletionIndex = 0
-                let previousResults = originalIncluded?[sectionIndex]
-                var sectionFilterResults = [Bool]()
-                var section = [Model]()
-                for (itemIndex, model) in sectionModels.enumerated() {
-                    var shouldInclude = originalFilter(model)
-                    if let limit = originalLimit , itemCount >= limit {
-                        shouldInclude = false
-                    }
-                    if shouldInclude {
+            for section in originalSections {
```

Comment:
would be nice to one day be able to check for cancellation during this tight loop in the async case. maybe add a TODO?